scripts/extensions/auto-tagging-widget/src/auto-tagging.tsx

```diff
@@ -0,0 +1,516 @@
+import {noop} from 'lodash';
+
+export const entityGroups = OrderedSet(['place', 'person', 'organisation']);
+
+export type INewItem = Partial<ITagUi>;
+
+interface IAutoTaggingResponse {
+    analysis: OrderedMap<string, ITagUi>;
+}
+
+interface IAutoTaggingSearchResult {
+    result: {
+        tags: IServerResponse;
+    };
+}
+
+interface IProps {
+    article: IArticle;
+}
+
+type IEditableData = {original: IAutoTaggingResponse; changes: IAutoTaggingResponse};
+
+interface IState {
+    runAutomaticallyPreference: boolean | 'loading';
+    data: 'not-initialized' | 'loading' | IEditableData;
+    newItem: INewItem | null;
+    vocabularyLabels: Map<string, string> | null;
+}
+
+const RUN_AUTOMATICALLY_PREFERENCE = 'run_automatically';
+
+function tagAlreadyExists(data: IEditableData, qcode: string): boolean {
+    return data.changes.analysis.has(qcode);
+}
+
+export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string) {
+    const {preferences} = superdesk;
+    const {httpRequestJsonLocal} = superdesk;
+    const {gettext} = superdesk.localization;
+    const {memoize, generatePatch} = superdesk.utilities;
+    const groupLabels = getGroups(superdesk);
+
+    const TagListComponent = getTagsListComponent(superdesk);
+    const NewItemComponent = getNewItemComponent(superdesk);
+
+    return class AutoTagging extends React.PureComponent<IProps, IState> {
+        private isDirty: (a: IAutoTaggingResponse, b: Partial<IAutoTaggingResponse>) => boolean;
+        private _mounted: boolean;
+
+        constructor(props: IProps) {
+            super(props);
+
+            this.state = {
+                data: 'not-initialized',
+                newItem: null,
+                runAutomaticallyPreference: 'loading',
+                vocabularyLabels: null,
+            };
+
+            this._mounted = false;
+            this.runAnalysis = this.runAnalysis.bind(this);
+            this.initializeData = this.initializeData.bind(this);
+            this.updateTags = this.updateTags.bind(this);
+            this.createNewTag = this.createNewTag.bind(this);
+            this.insertTagFromSearch = this.insertTagFromSearch.bind(this);
+            this.reload = this.reload.bind(this);
+            this.isDirty = memoize((a, b) => Object.keys(generatePatch(a, b)).length > 0);
+        }
+        runAnalysis() {
+            const dataBeforeLoading = this.state.data;
+
+            this.setState({data: 'loading'}, () => {
+                const {guid, language, headline, body_html} = this.props.article;
+
+                httpRequestJsonLocal<{analysis: IServerResponse}>({
+                    method: 'POST',
+                    path: '/ai/',
+                    payload: {
+                        service: 'imatrics',
+                        item: {
+                            guid,
+                            language,
+                            headline,
+                            body_html,
```

Comment:
can you add abstract here as well? or maybe just send the whole item. it's needed for SDNTB-667

Reply:
Will add after server changes in order not to break the testing instance.

Reply:
it's merged on the backend now